packages/builder/src/pages/builder/app/[application]/design/_components/NewScreen/CreateScreenModal.svelte

```diff
@@ -53,11 +52,13 @@
 
       for (let screen of screens) {
         // Check we aren't clashing with an existing URL
-        if (hasExistingUrl(screen.routing.route)) {
+        if (
+          screenStore.hasExistingUrl(screen.routing.route, screenAccessRole)
+        ) {
           let suffix = 2
-          let candidateUrl = makeCandidateUrl(screen, suffix)
-          while (hasExistingUrl(candidateUrl)) {
-            candidateUrl = makeCandidateUrl(screen, ++suffix)
+          let candidateUrl = screenStore.makeCandidateUrl(screen, suffix)
+          while (screenStore.hasExistingUrl(candidateUrl, screenAccessRole)) {
+            candidateUrl = screenStore.makeCandidateUrl(screen, ++suffix)
```

Comment:
Rather than all this logic (some of which I know was pre-existing), we should use our `getSequentialName` utility which is designed to do exactly this!